### ISSUE TITLE
Update ipykernel to >= 6.0

### ISSUE THREAD

**Issue by ptheywood** [googlecolab/colabtools#2230]
**Is your feature request related to a problem? Please describe.**
`ipykernel <= 6.0.0a4` does not show `stdout` or `stderr` within the notebook display. 
This was fixed by `ipykernel` `6.0.0a5`, speciifcally [ipython/ipykernel#630](https://github.com/ipython/ipykernel/pull/630)

**Describe the solution you'd like**
`ipykernel` to be updated to >= 6.0.0

**Describe alternatives you've considered**
Manually upgrading `ipykernel` within an active session generally works (after a kernel restart), but is understandably not recommended and will probably cause issues.

**Additional context**
Python bindings for a CUDA C++ library which uses NVRTC for runtime compilation of CUDA kernels outputs compialtion errors to `stdout`, which are not visible when using Colab (or other notebooks with `ipykernel < 6.0.0a5`

**Comment by cperry-goog** [comment]
This is on our radar, no specific timeline right now. Thanks.

**Comment by tburnett** [comment]
I found the need to update ipython, the default 5.5.0 failed to render my markdown output with HTML content. Requiring the update triggers the need to restart, an annoyance for me and my users.  (I also need an updated scipy, but that updates easily.)  This seems to be unrelated to the ipykernel version, but let me + this request to add recent versions of all the modules one expects in the jupyterlab environment--please remove this shadow on such a wonderful contribution to jupyter notebook-based collaboration.

**Comment by KevinGG** [comment]
+1 to upgrade ipykernel to v6 as it now supports Jupyter native debugger.

There is some non-backward change from ipython v6 to v7 that `display` package path is re-arranged. But once ipython is updated, both ipython v7 and v8 work with ipykernel v5+.

TL;DR: coming from https://github.com/apache/beam/issues/23599

**Comment by metrizable** [comment]
Colab uses `ipykernel==6.17.1` now.
